### PR TITLE
Generalize permissions boundary account id (#1561)

### PR DESCRIPTION
**Pull request by danielsotirhos**

### CODE REVIEW

File: src/azul/__init__.py [hunk 1]
```diff
@@ -162,6 +162,13 @@ def dss_direct_access(self) -> bool:
     def dss_direct_access_role(self, lambda_name: str) -> Optional[str]:
         return os.environ.get('AZUL_DSS_DIRECT_ACCESS_ROLE').format(lambda_name=lambda_name)
 
+    @property
+    def dss_direct_access_account(self) -> Optional[str]:
+        """
+        The direct access account id extracted from the direct access role value
+        """
+        return os.environ.get('AZUL_DSS_DIRECT_ACCESS_ROLE').split(':')[4]
```
Comment (hannes-ucsc) [inline src/azul/__init__.py:170]: What if `.get()` returns `None`?

Comment (hannes-ucsc) [reply src/azul/__init__.py:170]: @jessebrennan, this one is easy to catch, wouldn't you agree?

Comment (danielsotirhos) [reply src/azul/__init__.py:170]: The env variable `AZUL_DSS_DIRECT_ACCESS_ROLE` is already expected to be required by `dss_direct_access_role()`.
I agree however that if a key is required then `[]` should be used not `.get()`. I will update along with `dss_direct_access_role()` to match.

**Comment by danielsotirhos** [comment]
Upon closer inspection it seems like `config.dss_direct_access_role()` and `config.dss_direct_access_account` should only be required if direct access is enabled.

@jessebrennan  What are your thoughts on this approach with `_get_direct_access_related_value()`? I'm currently unsure if this approach is correct for the case where direct access is not enabled and the related variables (role & account) are defined in the environment, or if they should always be returned as None when direct access is not enabled.

File: src/azul/__init__.py (hunk 1 above)
Comment (hannes-ucsc) [reply src/azul/__init__.py]: Yes, it seems that I made the same mistake in config.dss_direct_access_role.

**Comment by hannes-ucsc** [comment]
Looks pretty complicated to me.

**Comment by danielsotirhos** [comment]
> Looks pretty complicated to me.

Simplified, though with the same assumptions. Let me know if you disagree that these config properties related to DSS direct access should only be required if DSS direct access is enabled.

File: src/azul/__init__.py (hunk 1 above)
Comment (jessebrennan) [reply src/azul/__init__.py]: Yes, I should have seen this.

**Pull request merged**